src/renderer/components/firmware/DownloadFirmwareButton.tsx

```diff
@@ -64,7 +74,52 @@ const DownloadFirmwareButton: React.FC<Props> = ({
   };
 
   const download = async (): Promise<void> => {
-    if (validPrVersion) {
+    // Try to get the cloudbuild build, if not found, throw an error
+    if (isCloudBuild && isCloudBuildValid) {
+      const flags = selectedFlags as { name: string; value: string }[];
+      const response = await client.query({
+        query: gql(/* GraphQL */ `
+          query CloudFirmwareStatus($params: CloudFirmwareParams!) {
+            cloudFirmwareStatus(params: $params) {
```

Comment:
Minor, but for GQL shapes you should consider just adding `status` as a key on the `cloudFirmware` query. Also the cloud firmware query should be queried by params, with `base64data` being another resolver which you query when the you want to download.

Reply:
If you wanna see what I mean, have a look at how `base64data` is queried in the github firmwares

Reply:
> Minor, but for GQL shapes you should consider just adding status as a key on the cloudFirmware resolver*

Don't really understand this part, why should status be a key on the resolver?
For the rest, now using downloadBinary directly.

Reply:
Nah, don't do it this way. What's the reason not to just provide the data as a resolver on the job status?

Reply:
IMO you should just make the structure of the resolver:
```
cloudFirmware(params: $params) {
  status
  base64firmware
}
```

Reply:
Basically, there shouldn't a separate `cloudFirmwareStatus` resolver. There should just be `status` resolver on the `cloudFirmware` resolver.

And also a `base64data` on that resolver too.